src/platforms/ios/iostunnel.swift

```diff
@@ -203,35 +205,43 @@ class PacketTunnelProvider: NEPacketTunnelProvider {
                 // Updates the tunnel configuration and responds with the active configuration
-                logger.info(message: "Switching tunnel configuration")
-
-                do {
-                  let tunnelConfiguration = try TunnelConfiguration(fromWgQuickConfig: configString)
-                  adapter.update(tunnelConfiguration: tunnelConfiguration) { error in
-                    if let error = error {
-                        self.logger.error(message: "Failed to switch tunnel configuration: \(error.localizedDescription)")
-                      completionHandler(nil)
-                      return
-                    }
-
-                    self.adapter.getRuntimeConfiguration { settings in
-                      var data: Data?
-                      if let settings = settings {
-                          data = settings.data(using: .utf8)!
-                      }
-                      completionHandler(data)
-                    }
-                  }
-                } catch {
-                  completionHandler(nil)
-                }
+                logger.info(message: "Switching tunnel configuration from app message")
+                updateServerConfig(with: configString, completionHandler: completionHandler)
             case .telemetryEnabledChanged(let uploadEnabled):
                 Glean.shared.setUploadEnabled(uploadEnabled)
                 completionHandler(nil)
+            case .silentServerSwitch:
+                silentServerSwitch()
+                completionHandler(nil)
             }
         } catch {
             logger.error(message: "Unexpected error while parsing message: \(error).")
             completionHandler(nil)
         }
     }
+
+    private func updateServerConfig(with configString: String, completionHandler: ((Data?) -> Void)? = nil) {
+        do {
+            let tunnelConfiguration = try TunnelConfiguration(fromWgQuickConfig: configString)
+            adapter.update(tunnelConfiguration: tunnelConfiguration) { error in
+                if let error = error {
+                    self.logger.error(message: "Failed to switch tunnel configuration: \(error.localizedDescription)")
+                    completionHandler?(nil)
+                    return
+                }
+            }
+        } catch {
+          completionHandler?(nil)
+        }
+    }
+
+    func silentServerSwitch() {
+        if let fallbackServerConfig = ((protocolConfiguration as? NETunnelProviderProtocol)?.providerConfiguration?["fallbackConfig"] as? String), !fallbackServerConfig.isEmpty {
+            self.logger.info(message: "Sending silent server switch in Network Extension")
+            updateServerConfig(with: fallbackServerConfig)
+            (protocolConfiguration as? NETunnelProviderProtocol)?.providerConfiguration?["fallbackConfig"] = ""
+        } else {
+            logger.info(message: "Silent server switch called, but no fallbackConfig available")
```

Comment:
nit: logger.error.

Reply:
It's not necessarily an error - 
1. In geos that only have one server, we expect it to be empty. 
2. If we've already used the fallback server, we'd also expect to be empty.

In light of that, do you think it's okay to keep it as an info?

Reply:
Yeap, that sounds reasonable.